Feedback Post Plugin

jhung opened the pull request:
A Wordpress plugin that allows feedback to be submitted as custom posts, and viewable by a custom user role.

jhung committed:
NO-JIRA: First commit of feedback plugin.

jhung committed:
NO-JIRA: Working archive pages. Added code comments.

jhung committed:
NO-JIRA: Updated README with setup instructions.

jhung committed:
NO-JIRA: Updated README, and fixed a problem with submission form being validated upon load.

jhung committed:
NO-JIRA: Added license and copyright information in README.

jhung committed:
NO-JIRA: Added nonce validation to form submission. Added security question customization to the README.

grrrero commented:
Hey @jhung! So I tested it on a different theme, and it looks good but just needs some spacing.
![screen shot 2017-08-17 at 13 12 06](https://user-images.githubusercontent.com/11557520/29424426-ba89d0c2-834d-11e7-923e-f0116b3efec5.png)

grrrero commented:
To compare with the default WP theme:
![screen shot 2017-08-17 at 13 23 05](https://user-images.githubusercontent.com/11557520/29424884-50f40d38-834f-11e7-9570-4cc2cd2055c7.png)

cindyli commented on feedback-post-admin-options.php:
```diff
@@ -0,0 +1,141 @@
+Callback function for the settings section. Currently this is empty as there
+is only 1 setting on the page and is fairly self descriptive.
+ */
+function feedback_post_plugin_settings_section_callback() {
```
Do we need this function that does nothing?

jhung replied:
That callback is required by [add_settings_section](https://codex.wordpress.org/Function_Reference/add_settings_section).

Instead we can pass it an anonymous empty function and delete feedback_post_plugin_settings_section_callback():

```
add_settings_section (
        'feedback-post-plugin-settings-section',
        'Settings for the Feedback Post Plugin',
        function() {},
        'feedback-post-plugin'
    );
```
Would that be better?

jhung replied:
I just tried replacing the callback with an anonymous function and it causes the page selector to not appear. So it would seem it's needed.

Looks like we will need to keep it, even though it is empty @cindyli.

cindyli commented on feedback-post-admin-options.php:
```diff
@@ -0,0 +1,141 @@
+            $error = "Invalid page type. Reverting to last known value. Please choose from the items in the list.";
+        }
+    } else {
+        $error = "Invalid page type. Reverting to last known value. Please choose from the items in the list.";
```
Shall this error be defined as a constant instead of being hardcoded here?

jhung committed:
NO-JIRA: defined constants for errors, and other cleanup.

jhung committed:
IDI-158: Added fields for handling unregistered businesses

Added a field to handle feedback for unregistered businesses on Big
Idea website.

jhung committed:
IDI-158: Added metadata to recipient options

The recipient options list now shows additional metadata. Added checks
for the Cimy user extra fields plugin.

jhung committed:
Fixed message for when no feedback available.

jhung committed:
Fix handling of form input and feedback post archive link

- “Other Information” text area now only saved if “Other” is chosen
from list of recipients.
- Getting the feedback archive now takes a string as an argument and
uses it as the name of the link.

jhung committed:
Help text and email notifications

Moved the help text for the Feedback form. Email is now sent to the
site administrator (specified in the Site Settings) whenever new
feedback is posted to the site.

jhung committed:
Fixed case where no Feedback is found. Updated email string.
